devices/HumanStateProvider/HumanStateProvider.cpp

```diff
@@ -257,8 +274,103 @@ class HumanStateProvider::impl
         iDynTree::VectorDynSize jointVelocities,
         iDynTree::Twist baseVelocity,
         std::unordered_map<std::string, iDynTree::Vector3>& linkAngularVelocityError);
+
+    // constructor
+    impl();
+};
+
+// ===============
+// RPC PORT PARSER
+// ===============
+
+class HumanStateProvider::impl::CmdParser : public yarp::os::PortReader
+{
+
+public:
+    std::atomic<bool> cmdStatus{false};
+    std::atomic<bool> cmdErease{false};
+    std::string parentLinkName;
+    std::string childLinkName;
+
+    void resetInternalVariables()
+    {
+        parentLinkName = "";
+        childLinkName = "";
+        cmdStatus = false;
+        cmdErease = false;
+    }
+
+    bool read(yarp::os::ConnectionReader& connection) override
+    {
+        yarp::os::Bottle command, response;
+        if (command.read(connection)) {
+
+            if (command.get(0).asString() == "help") {
+                response.addVocab(yarp::os::Vocab::encode("many"));
+                response.addString("The following commands can be used to apply a secondary calibration assuming the subject is in the zero configuration of the model for the calibrated links. \n"
+                                   "Enter <calibrate> to apply a secondary calibration for all the links \n"
+                                   "Enter <calibrate <linkName>> to apply a secondary calibration for the given link \n"
+                                   "Enter <calibrate <parentLinkName> <childLinkName>> to apply a secondary calibration for the given chain \n"
+                                   "Enter <reset <linkName>> to remove secondary calibration for the given link \n"
+                                   "Enter <reset> to remove all the secondary calibrations");
```

Comment:
I didn't test directly, but the same format is extensively used in robotolgy code (e.g. https://github.com/robotology/yarp/blob/4785668ae205f87a62c7777f77517c3b5558402c/src/libYARP_run/src/yarp/run/Run.cpp#L476-L478)

Reply:
In the link you provided there are multiple ``response.addString()``. I wonder whether that's the way to go.

Reply:
it should be the same since add string is just adding the string in a single string, anyway to be coherent with the structure usually used I switched to multiple `addString` https://github.com/robotology/human-dynamics-estimation/pull/164/commits/4b97b90002f4f89adf17d0da4d22a88dab382987